Programming Foundations: Algorithms/Factorial_test/lib/fact.rb

```diff
@@ -0,0 +1,22 @@
+module Factorial_test
+  class Fact
+    def initialize(n)
+      @n = n
+    end
+
+    def fact
+      if @n == 0
```

Comment:
We try to avoid many nested `if` for learning you have https://rubystyle.guide

You can also use rubocop to tell you what is bad styling in your code https://github.com/rubocop-hq/rubocop